open_spiel/games/phantom_go/phantom_go_board.h

```diff
@@ -0,0 +1,324 @@
+// In Virtual mode, an action (row, col) is row * 21 + col, and pass is 21*21+1.
+// All functions in this file (except stated otherwise) use these virtual
+// coordinates.
+//
+// However, in the OpenSpiel API (in go.{h, cc}), the actions are still exposed
+// as actions within 0, board_size*boardsize) (with pass = board_size **2.
+//
+// Normal go is standardly played on board of size 19, for Phantom Go, standard is size 9
+inline constexpr int kMaxBoardSize = 19;
+inline constexpr int kVirtualBoardSize = kMaxBoardSize + 2;
+inline constexpr int kVirtualBoardPoints =
+    kVirtualBoardSize * kVirtualBoardSize;
+
+using VirtualPoint = uint16_t;
+
+VirtualPoint VirtualPointFromBoardPoint(int boardPoint, int boardSize);
+int VirtualPointToBoardPoint(VirtualPoint p, int boardSize);
+
+inline constexpr VirtualPoint kInvalidPoint = 0;
+inline constexpr VirtualPoint kVirtualPass = kVirtualBoardPoints + 1;
+
+// Returns the VirtualPoint corresponding to the provided coordinates, e.g. "d4"
+// or "f10".
+VirtualPoint MakePoint(std::string s);
+
+// Converts a VirtualPoint to a string representation.
+std::string VirtualPointToString(VirtualPoint p);
+
+std::ostream &operator<<(std::ostream &os, VirtualPoint p);
+
+// Conversion functions between VirtualPoint and row/column representation.
+std::pair<int, int> VirtualPointTo2DPoint(VirtualPoint p);
+// Returns the point identifier in the Virtual 21*21 board from the (row, col)
+// 0-index coordinate in the concrete board.
+VirtualPoint VirtualPointFrom2DPoint(std::pair<int, int> row_col);
+
+// Converts an OpenSpiel action in range [0, board_size **2] to the
+// Virtual board range [0, kVirtualPass], and vice-versa.
+Action VirtualActionToAction(int virtual_action, int board_size);
+int ActionToVirtualAction(Action action, int board_size);
+
+/*std::string ActionToString(Action action, int board_size) {
+  return VirtualPointToString(ActionToVirtualAction(action, board_size));
+}*/
+
+// Returns a reference to a vector that contains all points that are on a board
+// of the specified size.
+const std::vector<VirtualPoint> &BoardPoints(int board_size);
+
+// To iterate over 4 neighbouring points, do
+//
+// VirtualPoint point;
+// for (auto p = Neighbours4(point); p; ++p) {
+//   // Do something on p..
+// }
+//
+class Neighbours4 {
+ public:
+  explicit Neighbours4(const VirtualPoint p);
+
+  Neighbours4 &operator++();
+  const VirtualPoint operator*() const;
+  explicit operator bool() const;
+
+  
+
+ private:
+  VirtualPoint dir_;
+  const VirtualPoint p_;
+};
+
+// Simple Go board that is optimized for speed.
+// It only implements the minimum of functionality necessary to support the
+// search and is optimized for speed and size. Importantly, it fits on the
+// stack. For detailed numbers, run the benchmarks in go_board_test.
+class PhantomGoBoard {
+ public:
+  explicit PhantomGoBoard(int board_size);
+
+  void Clear();
+
+  
+
+  std::array<int, 2> getStoneCount() const { return stone_count_; };
```

Comment:
Change the method to `GetStoneCount(GoColor color)` so you avoid the uint8 casting.